content/blog/host-based-volume-encryption-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,249 @@
+
+# Inspecting the wire
+
+Assume we would write "Hello World" into an unencrypted volume versus an encrypted. In the following experiment I captured `echo "Hello World" > /data/file.txt && sync` with a network packet sniffer and loaded up the result in Wireshark.
+
+![Hello World unencrypted](/img/unecrypted.png "Hello World unencrypted")
+
+Imagine if this would be Personal Identifiable Information (PII) traveling across a compromised network. A security breach could potentially result in an expensive lawsuit if the information end up in the wrong hands. Let's repeat the experiment with an encrypted device.
+
+![Hello World encrypted](/img/encrypted.png "Hello World encrypted")
+
+Inspecting the same packet in the sequence, it's quite obvious the entire payload has been encrypted. This, of course, has impact on the actual storage. Data you know traditionally compress and deduplicate well, all of a sudden present a near 1:1 representation.
+
+I conducted an experiment where I wrote an 8GiB file, filled with zeroes. Under normal circumstances, such a file is nearly invisible to any enterprise array. Not when encrypted.
+
+```markdown
+HPE Alletra $ vol --info  pvc-44159f94-0f2b-4109-ae90-e48d0df082b1 | grep ^Volume
+Volume mapped usage (MiB): 7859
+Volume compression: 0.97X
+```
+
+It's advised to consult with the storage administrator to ensure data-at-rest encryption on the array is turned off along with deduplication and compression to spare CPU cycles.
+
+Naturally, there will be a performance impact on the host and it's advised to study empirical data from benchmarks conducted with a production-like workload to understand the amount of CPU overhead is needed to ensure the application meets its performance criteria.
+
+# Maliciously trying to retrieve data
+
+In the event of a rogue host gain access to a volume without having the key, assume the LUN gets connected and discovered.
+
+```bash
+# mount /dev/mapper/mpatha /mnt
+mount: unknown filesystem type 'crypto_LUKS'
+```
+
+The encrypted volume may in this state be brute forced or maliciously deleted, it's advisable to use a strong non-dictionary passphrase to encrypt and decrypt with. The passphrase can be up to 512 characters long. The passphrase length does not affect performance or the cipher strength, it's only used to open the device to the host.
+
+As an extra layer of security when using the iSCSI protocol is to use the Challenge-Handshake Authentication Protocol (CHAP) facility available to the HPE CSI Driver, that ensures a mutual (between initiator and target) shared secret is needed to perform a discovery in the first place.
```

Comment:
Recommend rephrasing to: "As an extra layer of security when using the iSCSI protocol, you can use the Challenge-Handshake Authentication Protocol (CHAP) facility available to the HPE CSI Driver. This ensures a mutual (between initiator and target) shared secret is needed to perform a discovery in the first place."